src/main/java/com/connexta/store/controllers/StoreController.java

```diff
@@ -93,7 +91,8 @@ public StoreController(@NotNull final StoreService storeService) {
   }
 
   @Override
-  public @NotNull ResponseEntity<Resource> retrieveProduct(@NotBlank String productId) {
+  public @NotNull ResponseEntity<Resource> retrieveProduct(
```

Comment:
The @NotNull and @NotBlank annotations caused an org.apache.http.client.ClientProtocolException and had to be removed.